### ISSUE TITLE
Problem with circular deps in built version

### ISSUE THREAD

**Issue by whitecolor** [stealjs/steal-tools#323]
So I have two modules

models/user.js: 
```javascript
import Purchase from './purchase'`

var User = can.Model.extend(...)

export default User
```

models/purchase.js:
```javascript
import User from './user'`

var Purchase = can.Model.extend(...)

export default Purchase
```

It all works in dev version, but after build

in say Purchase module User is imported ok, but in User imported Purchase is empty object {}

What can be the reason?

**Comment by matthewp** [comment]
ES6 and AMD have different ways to do circular references, that could be the reason. Changing those to non-es6 modules might fix it. If this is the reason (not positive) we could have an option to transpile to the `register` format that I believe does circular refs the es6 way.

**Comment by whitecolor** [comment]
Well I don't want to get rid of ES6 modules, currently as workaround I got rid of circular deps, but that is not a solution.

**Comment by matthewp** [comment]
Yeah, I was suggested it as a workaround not a solution. Need a test to see what the problem is.

**Comment by matthewp** [comment]
Another solution would be to detect the circular references and fix them when we transpile.

This might work as a workaround for you:

```js
import * as PurchaseModule from './purchase'`

setTimeout(function(){
  var Purchase = PurchaseModule["default"];
});
```

**Comment by whitecolor** [comment]
Actually I could n't reproduce it on small sample such example with ES6 modules seem to work
https://github.com/ModuleLoader/es6-module-loader/wiki/Circular-References-&-Bindings

**Comment by whitecolor** [comment]
BTW, I belive I experienced issues with circular deps with CJS too, so have to get rid of them.

**Comment by matthewp** [comment]
ES6 circular deps are fixed in 1.0
